base/uk.ac.stfc.isis.ibex.ui.synoptic.editor/src/uk/ac/stfc/isis/ibex/ui/synoptic/editor/target/selector/TargetSelectorPanel.java

```diff
@@ -0,0 +1,228 @@
+
+    private Map<String, List<String>> availableOpis;
+    private final TargetSelectorViewModel viewModel;
+    private Text txtName;
+    private Text txtDescription;
+    private ComboViewer comboType;
+    private Text txtSelectedTarget;
+
+    /**
+     * @param parent
+     *            the parent composite
+     * @param style
+     *            the style of this composite
+     * @param synopticViewModel
+     *            the synoptic view model
+     */
+    public TargetSelectorPanel(Composite parent, int style, final SynopticViewModel synopticViewModel) {
+        super(parent, style);
+
+        this.viewModel = new TargetSelectorViewModel(synopticViewModel);
+        this.availableOpis = viewModel.getAvailableOpis();
+
+        setLayout(new GridLayout(1, false));
+        setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+
+        // Draw the components
+        drawControls();
+        drawTargetTree();
+        drawDescriptionBox();
+
+        viewModel.addPropertyChangeListener("enabled", new PropertyChangeListener() {
+            @Override
+            public void propertyChange(PropertyChangeEvent evt) {
+                Utils.recursiveSetEnabled(TargetSelectorPanel.this, viewModel.isEnabled()); 
+            }
+        });
+        
+        // Have to do this here otherwise the binding overrides the default value in the model.
+        Utils.recursiveSetEnabled(this, viewModel.isEnabled());
+        
+        bind();
+    }
+
+    @SuppressWarnings("checkstyle:magicnumber") // Gui method   
+    private void drawControls() {
+
+        Composite container = new Composite(this, SWT.NONE);
+        container.setLayout(new GridLayout(3, false));
+        container.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
+
+        Label lblName = new Label(container, SWT.NONE);
+        lblName.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false));
+        lblName.setText("Name: ");
+
+        txtName = new Text(container, SWT.BORDER);
+        txtName.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 2, 1));
+        txtName.addVerifyListener(new ComponentNameVerifier());
+
+        Label lblIcon = new Label(container, SWT.NONE);
+        lblIcon.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false));
+        lblIcon.setText("Icon: ");
+
+        comboType = new ComboViewer(container);
+        comboType.getCombo().setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 2, 1));
+        comboType.getCombo().setItems(viewModel.componentTypesArray);
+        
+        Label lblSelectedTarget = new Label(container, SWT.NONE);
+        lblSelectedTarget.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false));
+        lblSelectedTarget.setText("Target OPI: ");
+
+        txtSelectedTarget = new Text(container, SWT.BORDER | SWT.READ_ONLY);
+        txtSelectedTarget.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
```

Comment:
I know I told you to put this on top of the OPI selection panel, however having had another look, the OPI name and description fields should really be next to each other. If you want we can have another look at it together to decide what would be the neatest solution.